pkg/github/github.go

```diff
@@ -476,6 +492,51 @@ func (g *GitHubClient) RepoUserPermissionLevel(ctx context.Context, owner, repo,
 	return permissionLevel, nil
 }
 
+func (g *GitHubClient) ListJobsForWorkflowRun(ctx context.Context, owner, repo string, runID int64, opts *github.ListWorkflowJobsOptions) (*JobsResponse, error) {
+	var jobs []*Job
+	var pagination *Pagination
+
+	if err := g.withRetries(ctx, func(ctx context.Context) error {
+		ghJobs, resp, err := g.client.Actions.ListWorkflowJobs(ctx, owner, repo, runID, opts)
+		if err != nil {
+			if resp != nil {
+				if _, ok := ignoredStatusCodes[resp.StatusCode]; !ok {
+					return retry.RetryableError(err)
+				}
+			}
+			return fmt.Errorf("failed to list jobs for workflow run attempt: %w", err)
+		}
+
+		for _, workflowJob := range ghJobs.Jobs {
+			jobs = append(jobs, &Job{ID: workflowJob.GetID(), Name: workflowJob.GetName()})
+		}
+
+		if resp.NextPage != 0 {
+			pagination = &Pagination{NextPage: resp.NextPage}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to list jobs for workflow run attempt: %w", err)
+	}
+
+	return &JobsResponse{Jobs: jobs, Pagination: pagination}, nil
+}
+
+func (g *GitHubClient) ResolveJobLogsURL(ctx context.Context, jobName, serverURL, owner, repo string, runID int64) (string, error) {
+	jobs, err := g.ListJobsForWorkflowRun(ctx, owner, repo, runID, nil)
```

Comment:
Looks like this response returns an `html_url` field, maybe we should use that instead of generating it ourselves?

https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run

Reply:
Oops, good catch! Done!